src/main/java/com/minecolonies/coremod/entity/ai/basic/AbstractEntityAIRequestSmelter.java

```diff
@@ -423,6 +434,31 @@ private BlockPos getPositionOfOvenToRetrieveFrom()
         return null;
     }
 
+    /**
+     * Get the furnace which has used fuel. For this check each furnace which has been registered to the building. Check if the furnace has used fuel in the fuel slot.
+     *
+     * @return the position of the furnace.
+     */
+    protected BlockPos getPositionOfOvenToRetrieveFuelFrom()
+    {
+        for (final BlockPos pos : getOwnBuilding().getFurnaces())
+        {
+            final TileEntity entity = world.getTileEntity(pos);
+            if (entity instanceof FurnaceTileEntity)
+            {
+                final FurnaceTileEntity furnace = (FurnaceTileEntity) entity;
+
+                if (furnace.getStackInSlot(FUEL_SLOT).getItem() == Items.BUCKET)
```

Comment:
While bucket is the only 'valid' result today, I can see the possibility of other items in a modded world. This might be better as "if the stack in the slot isn't a valid fuel"

Reply:
I was wondering if this would be better as "if the stack in the slot isn't an allowed fuel"

Reply:
So any fuel that is turned off in the GUI will be removed from the fuel slot as well

Reply:
mmm, i believe (not certain) "Lava Cells" from industrial craft 2 would be an example

Reply:
OK done.